core/src/main/java/org/apache/druid/java/util/metrics/JvmMonitor.java

```diff
@@ -162,224 +143,184 @@ private void emitDirectMemMetrics(ServiceEmitter emitter)
 
       Map<String, String[]> dimensionsCopy = dimensionsCopyBuilder.build();
 
       if (collector != null) {
         collector.emit(emitter, dimensionsCopy);
       }
 
-      for (GcGenerationSpace space : spaces) {
-        space.emit(emitter, dimensionsCopy);
-      }
     }
   }
 
   private class GcGenerationCollector
   {
-    private final String name;
-    private final LongCounter invocationsCounter;
-    private final TickCounter cpuCounter;
     private long lastInvocations = 0;
     private long lastCpuNanos = 0;
+    private final GarbageCollectorMXBean gcBean;
 
-    GcGenerationCollector(Map<String, JStatData.Counter<?>> jStatCounters, long genIndex)
+    GcGenerationCollector(GarbageCollectorMXBean gcBean)
     {
-      String collectorKeyPrefix = StringUtils.format("sun.gc.collector.%d", genIndex);
-
-      String nameKey = StringUtils.format("%s.name", collectorKeyPrefix);
-      StringCounter nameCounter = (StringCounter) jStatCounters.get(nameKey);
-      name = getReadableName(nameCounter.getString());
-
-      invocationsCounter = (LongCounter) jStatCounters.get(StringUtils.format("%s.invocations", collectorKeyPrefix));
-      cpuCounter = (TickCounter) jStatCounters.get(StringUtils.format("%s.time", collectorKeyPrefix));
+      this.gcBean = gcBean;
     }
 
     void emit(ServiceEmitter emitter, Map<String, String[]> dimensions)
     {
       final ServiceMetricEvent.Builder builder = builder();
       MonitorUtils.addDimensionsToBuilder(builder, dimensions);
 
-      long newInvocations = invocationsCounter.getLong();
+      long newInvocations = gcBean.getCollectionCount();
       emitter.emit(builder.build("jvm/gc/count", newInvocations - lastInvocations));
       lastInvocations = newInvocations;
 
-      long newCpuNanos = cpuCounter.getNanos();
+      long newCpuNanos = gcBean.getCollectionTime();
       emitter.emit(builder.build("jvm/gc/cpu", newCpuNanos - lastCpuNanos));
       lastCpuNanos = newCpuNanos;
     }
+  }
 
-    private String getReadableName(String name)
+  private class GcSpaceCollector
+  {
+
+    private final List<GcGenerationSpace> spaces = new ArrayList<>();
+
+    public GcSpaceCollector(MemoryPoolMXBean memoryPoolMxBean)
     {
-      switch (name) {
-        // Young gen
-        case "Copy":
-          return "serial";
-        case "PSScavenge":
-          return "parallel";
-        case "PCopy":
-          return "cms";
-        case "G1 incremental collections":
-          return "g1";
-        case "Shenandoah partial":
-          return "shenandoah";
-
-        // Old gen
-        case "MCS":
-          return "serial";
-        case "PSParallelCompact":
-          return "parallel";
-        case "CMS":
-          return "cms";
-        case "G1 stop-the-world full collections":
-          return "g1";
-        case "Shenandoah full":
-          return "shenandoah";
+      MemoryUsage collectionUsage = memoryPoolMxBean.getCollectionUsage();
+      if (collectionUsage != null) {
```

Comment:
I think we can get `MemoryUsage` object out of this ctor and check it if is null before constructing this `GcSpaceCollector` object. In this way, if the `collectionUsage` is not null, there is no `GcSpaceCollector` object creation.